ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -1405,13 +1413,27 @@ private System.Collections.IEnumerator
 
   private void Early() {
     if (PluginRunning()) {
+      // Wait for all the asynchronous integrations to complete and kill the
+      // vessels that collided with a celestial.  Note that a given vessel may
+      // be returned by several calls to FutureWaitForVesselToCatchUp because
+      // what we integrate are really pile-ups.
+      var all_collided_vessels = new HashSet<Vessel>();
       foreach (var f in vessel_futures_) {
         var future = f;
         IntPtr collided_vessels;
         plugin_.FutureWaitForVesselToCatchUp(ref future,
                                              out collided_vessels);
+        for (; !collided_vessels.IteratorAtEnd();
+             collided_vessels.IteratorIncrement()) {
+          Guid vessel_guid = new Guid(collided_vessels.IteratorGetVesselGuid());
+          Vessel vessel = FlightGlobals.FindVessel(vessel_guid);
```

Comment:
Handle the case where FindVessel returns null.

Reply:
Done.